nexus/lib/qmcpack_input.py

```diff
@@ -5254,6 +5255,57 @@ def generate_jastrow3(function='polynomial',esize=3,isize=3,rcut=4.,coeff=None,i
 #end def generate_jastrow3
 
 
+def generate_kspace_jastrow(kc1, kc2, nk1, nk2,
+  symm1='isotropic', symm2='isotropic', coeff1=None, coeff2=None):
+  """ generate <jastrow type="kSpace">
+
+  Args:
+    kc1 (float): kcut for one-body Jastrow
+    kc2 (float): kcut for two-body Jastrow
+    nk1 (int): number of coefficients for one-body Jastrow
+    nk2 (int): number of coefficients for two-body Jastrow
+    symm1 (str, optional): one of ['crystal', 'isotropic', 'none'],
+     default is 'isotropic'
+    symm2 (str, optional): one of ['crystal', 'isotropic', 'none'],
+     default is 'isotropic'
+    coeff1 (list, optional): one-body Jastrow coefficients, default None
+    coeff2 (list, optional): two-body Jastrow coefficients, default None
+  Returns:
+    QIxml: kspace_jastrow qmcpack_input element
+  """
+
+  if coeff1 is None: coeff1 = [0]*nk1
+  if coeff2 is None: coeff2 = [0]*nk2
+  if len(coeff1) != nk1: raise RuntimeError('coeff1 mismatch')
+  if len(coeff2) != nk2: raise RuntimeError('coeff2 mismatch')
+
+  corr1 = correlation({
+    'type': 'One-Body',
+    'symmetry': symm1,
+    'kc': kc1,
+    'coefficients': section({
+      'id': 'cG1', 'type': 'Array',
+      'coeff': coeff1
+     })
+  })
```

Comment:
This looks a little heavy syntactically.  Does this not work:

  corr1 = correlation(
    type = 'One-Body',
    symmetry = symm1,
    kc = kc1,
    coefficients = section(
      id = 'cG1', type = 'Array',
      coeff = coeff1
     )
  )

I generally prefer this form for Nexus since it removes two layers of markup: one for the dict "{}" and one for each key "''".

Reply:
I use dictionary in my own scripts because then I can pass the dictionary around to other simple scripts that know nothing about nexus. Now, since this function is being added to nexus, I agree that the internal syntax is cleaner.